types/stripe/index.d.ts

```diff
@@ -5206,7 +5208,1262 @@ declare namespace Stripe {
+				 * The amount Stripe held from your account to fund the authorization, if the request was approved
+				 */
+				held_amount: number;
+
+				/**
+				 * The currency of the held amount
+				 */
+				held_currency: string;
+
+				/**
+				 * One of authentication_failed, authorization_controls, card_active, card_inactive, insufficient_funds, account_compliance_disabled, account_inactive, suspected_fraud, webhook_approved, webhook_declined, or webhook_timeout.
+				 */
+				reason: 'authentication_failed' | 'authorization_controls' | 'card_active' | 'card_inactive' | 'insufficient_funds' | 'account_compliance_disabled' | 'account_inactive' | 'suspected_fraud' | 'webhook_approved' | 'webhook_declined' | 'webhook_timeout';
+
+				/**
+				 * When an authorization is declined due to authorization_controls, this array contains details about the authorization controls that were violated. Otherwise, it is empty.
+				 */
+				violated_authorization_controls: {
+					/**
+					 * Entity which the authorization control acts on. One of account, card, or cardholder.
+					 */
+					entity: 'account' | 'cardholder' | 'card',
+
+					/**
+					 * Name of the authorization control. One of allowed_categories, blocked_categories, max_amount, max_approvals, or spending_limits.
+					 */
+					name: 'allowed_categories' | 'blocked_categories' | 'max_amount' | 'max_approvals' | 'spending_limits',
+				};
+			}
+
+			interface VerificationData {
+				/**
+				 * One of match, mismatch, or not_provided.
+				 */
+				address_line1_check: 'match' | 'mismatch' | 'not_provided';
+
+				/**
+				 * One of match, mismatch, or not_provided.
+				 */
+				address_zip_check: 'match' | 'mismatch' | 'not_provided';
+
+				/**
+				 * One of exempt, failure, none, or success.
+				 */
+				authentication: 'exempt' | 'failure' | 'none' |'success';
+
+				/**
+				 * One of match, mismatch, or not_provided.
+				 */
+				cvc_check: 'match' | 'mismatch' | 'not_provided';
+			}
+
+			type AuthorizationStatus = 'closed' | 'pending' | 'reversed';
+			type AuthorizationMethod = 'chip' | 'contactless' | 'keyed_in' | 'online' | 'swipe';
+			type WalletProvider = 'apple_pay' | 'google_pay' | 'samsung_pay';
+
+			interface IAuthorizationUpdateOptions {
+				/**
+				 * Set of key-value pairs that you can attach to an object. This can be useful for storing additional information about the object in a structured format.
+				 */
+				metadata?: IMetadata;
+			}
+
+			interface IAuthorizationApproveOptions {
+				/**
+				 * If the authorization’s is_held_amount_controllable property is true, you may provide this value to control how much to hold for the authorization.
+				 * Must be positive (use decline to decline an authorization request).
+				 */
+				held_amount?: number;
+
+				/**
+				 * Set of key-value pairs that you can attach to an object. This can be useful for storing additional information about the object in a structured format.
+				 */
+				metadata?: IMetadata;
```

Comment:
Options classes should use `IOptionsMetadata` instead of `IMetadata`.

Reply:
@pushplay resolved. Let me know how it looks now.